### PR TITLE
Document new 'register_type' parameter for modbus binary sensor

### PR DESCRIPTION
**Pull request by keton**
**Description:**

**Pull request in home-assistant (if applicable):** home-assistant/home-assistant#27397

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next Home Assistant release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_integrations/binary_sensor.modbus.markdown [hunk 1]
```diff
@@ -27,6 +27,7 @@ binary_sensor:
         hub: hub1
         slave: 1
         coil: 110
+        register_type: discrete_input
```
Comment (frenck) [inline source/_integrations/binary_sensor.modbus.markdown]: We try to keep all configuration samples minimal. Thus, no optional requirement in the default sample. This helps a user to get started quickly by copy-&-paste the sample without worrying about optional parameters which they most likely not need. If required, insert a full configuration sample later that covers special setups or alike.

**Review by frenck — CHANGES_REQUESTED** [review]
Thanks for the PR @keton! 

I've reviewed it and left a comment. Could you please take a look? Thanks! 👍

File: source/_integrations/binary_sensor.modbus.markdown (hunk 1 above)
Comment (keton) [reply source/_integrations/binary_sensor.modbus.markdown]: Thanks for the clarification, offending line has been removed

**Review by frenck — APPROVED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

**Pull request closed without merging**